modules/apps/object/object-dynamic-data-mapping/src/main/resources/META-INF/resources/Autocomplete.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {Autocomplete as BaseAutocomplete} from 'frontend-js-components-web';
+import {debounce} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+function getData(apiUrl, query, page, pageSize) {
+	const url = new URL(apiUrl, Liferay.ThemeDisplay.getPortalURL());
+
+	if (query) {
+		url.searchParams.append('search', query);
+	}
+
+	if (page) {
+		url.searchParams.append('page', page);
+	}
+
+	if (pageSize) {
+		url.searchParams.append('pageSize', pageSize);
+	}
```

Comment:
Just a comment here about the use of `append` vs `set`. Doing a quick grep in our codebase I see:
- `searchParams.set`: 40 results in 28 files
- `searchParams.append`: 62 results in 24 files

I bring this up because `append` will always append a new key-value pair regardless of it existing it previously in the URL. 

So, if the url happened to be: `foo.com?page=1`, then `append('page', 2)` would generate `foo.com?page=1&page=2`. It would be up to the server then to handle `page` as a list param, but more often than not, first one will win and the effect of setting the param page to 2 would be negated.

I think in this case it's fairly safe because we are more or less certain `themeDisplay.getPortalURL` doesn't have search params and we're not passing this url around where others could manipulate it.

I'm not sure what, or how we'd want to do about it, but I get the feeling we're not always being 100% intentional when using these APIs, what could result in hard-to-diagnose issues in the future.